data-source/aws_vpc: TestAccDataSourceAwsVpc_multipleCidr Flaky

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

The `TestAccDataSourceAwsVpc_multipleCidr` acceptance test has inconsistent failures such as:

```
=== CONT  TestAccDataSourceAwsVpc_multipleCidr
data_source_aws_vpc_test.go:118: Step 1/1 error: Error running apply: exit status 1
Error: multiple VPCs matched; use additional constraints to reduce matches to a single VPC
--- FAIL: TestAccDataSourceAwsVpc_multipleCidr (46.18s)
```

The data source configuration it uses is dependent on filtering by `cidr-block-association.cidr-block`, which can be in use by many of the other testing VPCs. Rather than using that filter, it can instead use a direct identifier lookup since the purpose of the test is not to verify that filters work, but that multiple CIDR block associations are returned as expected in the data source.

### Affected Resource(s)

* aws_vpc

### Potential Terraform Configuration

```hcl
data "aws_vpc" "test" {
  id = aws_vpc_ipv4_cidr_block_association.test.vpc_id
}
```

--- a/aws/data_source_aws_vpc_test.go
+++ b/aws/data_source_aws_vpc_test.go
@@ -11,8 +11,8 @@ import (
 
 func TestAccDataSourceAwsVpc_basic(t *testing.T) {
 	rand.Seed(time.Now().UTC().UnixNano())
-	rInt := rand.Intn(16)
-	cidr := fmt.Sprintf("172.%d.0.0/16", rInt)
+	rInt := rand.Intn(254)
+	cidr := fmt.Sprintf("10.%d.0.0/16", rInt+1) // Prevent common 10.0.0.0/16 cidr_block matches
 	tag := fmt.Sprintf("terraform-testacc-vpc-data-source-basic-%d", rInt)
 
 	vpcResourceName := "aws_vpc.test"
@@ -79,8 +79,8 @@ func TestAccDataSourceAwsVpc_basic(t *testing.T) {
 
 func TestAccDataSourceAwsVpc_ipv6Associated(t *testing.T) {
 	rand.Seed(time.Now().UTC().UnixNano())
-	rInt := rand.Intn(16)
-	cidr := fmt.Sprintf("172.%d.0.0/16", rInt)
+	rInt := rand.Intn(255)
+	cidr := fmt.Sprintf("10.%d.0.0/16", rInt)
 	tag := fmt.Sprintf("terraform-testacc-vpc-data-source-ipv6-associated-%d", rInt)
 
 	vpcResourceName := "aws_vpc.test"
@@ -111,9 +111,8 @@ func TestAccDataSourceAwsVpc_ipv6Associated(t *testing.T) {
 	})
 }
 
-func TestAccDataSourceAwsVpc_multipleCidr(t *testing.T) {
-	rInt := rand.Intn(16)
-	rName := "data.aws_vpc.test"
+func TestAccDataSourceAwsVpc_CidrBlockAssociations_Multiple(t *testing.T) {
+	dataSourceName := "data.aws_vpc.test"
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
@@ -121,9 +120,9 @@ func TestAccDataSourceAwsVpc_multipleCidr(t *testing.T) {
 		CheckDestroy: testAccCheckVpcDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDataSourceAwsVpcConfigMultipleCidr(rInt),
+				Config: testAccDataSourceAwsVpcConfigCidrBlockAssociationsMultiple(),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(rName, "cidr_block_associations.#", "2"),
+					resource.TestCheckResourceAttr(dataSourceName, "cidr_block_associations.#", "2"),
 				),
 			},
 		},
@@ -173,29 +172,26 @@ data "aws_vpc" "by_tag" {
 
 data "aws_vpc" "by_filter" {
   filter {
-    name   = "cidr"
-    values = [aws_vpc.test.cidr_block]
+    name   = "vpc-id"
+    values = [aws_vpc.test.id]
   }
 }
 `, cidr, tag)
 }
 
-func testAccDataSourceAwsVpcConfigMultipleCidr(octet int) string {
-	return fmt.Sprintf(`
+func testAccDataSourceAwsVpcConfigCidrBlockAssociationsMultiple() string {
+	return `
 resource "aws_vpc" "test" {
-  cidr_block = "10.%d.0.0/16"
+  cidr_block = "10.0.0.0/16"
 }
 
 resource "aws_vpc_ipv4_cidr_block_association" "test" {
   vpc_id     = aws_vpc.test.id
-  cidr_block = "172.%d.0.0/16"
+  cidr_block = "172.0.0.0/16"
 }
 
 data "aws_vpc" "test" {
-  filter {
-    name   = "cidr-block-association.cidr-block"
-    values = [aws_vpc_ipv4_cidr_block_association.test.cidr_block]
-  }
+  id = aws_vpc_ipv4_cidr_block_association.test.vpc_id
 }
-`, octet, octet)
+`
 }

